version: '3'

services:

  redis:
    image: redis
    ports:
      - '6379:6379'

  db:
    image: postgres
    env_file:
      - .env
    environment:
        - POSTGRES_USER=${DATABASES_USER}
        - POSTGRES_PASSWORD=${DATABASES_PASSWORD}
        - POSTGRES_DB=${DATABASES_NAME}
        - PGDATA=/var/lib/postgresql/data/pgdata

    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -d $${POSTGRES_DB} -U $${POSTGRES_USER}" ]
      interval: 10s
      timeout: 5s
      retries: 5

    volumes:
      - pg_data:/var/lib/postgresql/data/pgdata
    ports:
      - '5432:5432'

  app:
    build: .
    tty: true
    command: sleep 3 && python3 manage.py migrate && python3 manage.py runserver 0.0.0.0:8007

    ports:
      - '8007:8007'
    env_file:
      - .env
    volumes:
      - .:/code



  celery:
    build: .
    tty: true
    env_file:
      - .env
    command: sleep 10 && celery -A config worker -l INFO
    depends_on:
      - redis
      - app


  celery_beat:
    build: .
    tty: true
    env_file:
      - .env
    command: sleep 11 && celery -A config beat -l INFO
    depends_on:
      - redis
      - app


volumes:
  pg_data:
